src/app/company/state/company.state.ts

```diff
@@ -0,0 +1,16 @@
+import { CompanyItem } from '../models/company.model';
+import { LocalizedErrorInfo } from 'src/app/shared/models/error-info.model';
+
+export interface CompanyState {
+    companies: CompanyItem[];
+    currentCompany: CompanyItem | null;
```

Comment:
currentCompany by mohlo byť v store uložené iba ako currentCompanyId - lebo teraz je tá istá entita niekde v kolekcii companies aj v currentCompany.
Fungovanie zvyšku appky by to nezmenilo, lebo stačí urobiť selector na getCurrentCompany, ktorý ju dohľadá v companies v store pomocou kompozície selektorov...